### PR TITLE
Add .swiftlint.yml

### LINKED ISSUE #382: [discussion] Exclude `Test` folder on SwiftLint.
There is a lot of violations in test sources (e.g. https://github.com/danger/swift/pull/381#discussion_r501453860).
I counted violations with [this script](https://gist.github.com/417-72KI/75fbe862ae5645b7fea67e2645eecdda), and here is a result.

```
1 file_length
1 todo
2 force_try
2 xctfail_message
3 function_body_length
3 type_body_length
7 trailing_whitespace
57 identifier_name
73 trailing_comma
142 line_length
```

When `Tests` is excluded on SwiftLint, the result will be as follows.

```
1 force_try
1 todo
1 trailing_whitespace
3 line_length
4 identifier_name
```

From the above results, we can see that most of the violations are in the test code.

I think violations from SwiftLint should be fixed on source codes, but can be ignored on test codes.

### PR DESCRIPTION
**Pull request by 417-72KI**
resolves #382

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+line_length:
+  warning: 140 # needs discussion (default: 120)
+  error: 350 # needs discussion (default: 200)
+  ignores_comments: true
+  ignores_urls: true
+  ignores_function_declarations: true
+  ignores_interpolated_strings: true
+identifier_name:
+  min_length: 
+    warning: 2
```
Comment (f-meloni) [inline .swiftlint.yml:16]: Can we use `excluded:` to add the excluded words instead? (like id)

**Review by f-meloni — APPROVED** [review]

**Comment by f-meloni** [comment]
Thank you very much :)

**Comment by f-meloni** [comment]
PS. I believe you will need to add the config file to the Dangerfile https://github.com/danger/swift/blob/master/Dangerfile.swift#L23

File: .swiftlint.yml (hunk 1 above)
Comment (417-72KI) [reply .swiftlint.yml]: Sure! fixed in 38131f0

**Comment by danger-public** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/RunnerLib/NSRegularExpression+FilesImport.swift#L6** - Force tries should be avoided. (`force_try`)
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  **Sources/Danger/DangerUtils.swift#L112** - Line should be 120 characters or less: currently 141 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/CliArgs/CliArgs.swift#L4** - Variable name should be between 3 and 40 characters long: 'id' (`identifier_name`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/CliArgs/CliArgs.swift#L22** - Enum element name should be between 3 and 40 characters long: 'id' (`identifier_name`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/DangerCommand.swift#L2** - Enum element name should be between 3 and 40 characters long: 'ci' (`identifier_name`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/DangerCommand.swift#L4** - Enum element name should be between 3 and 40 characters long: 'pr' (`identifier_name`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/NSRegularExpression+FilesImport.swift#L6** - Line should be 120 characters or less: currently 124 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/Runtime.swift#L38** - TODOs should be resolved (Support running from a fork?). (`todo`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/SPMDanger.swift#L53** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 652ee85e4214a7f0bce6d812b79a14bf6517ab90
</p>

**Comment by danger-public** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/RunnerLib/NSRegularExpression+FilesImport.swift#L6** - Force tries should be avoided. (`force_try`)
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  **Sources/Danger/DangerUtils.swift#L112** - Line should be 120 characters or less: currently 141 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/DangerDependenciesResolver/Package.swift#L40** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/CliArgs/CliArgs.swift#L4** - Variable name should be between 3 and 40 characters long: 'id' (`identifier_name`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/CliArgs/CliArgs.swift#L22** - Enum element name should be between 3 and 40 characters long: 'id' (`identifier_name`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/DangerCommand.swift#L2** - Enum element name should be between 3 and 40 characters long: 'ci' (`identifier_name`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/DangerCommand.swift#L4** - Enum element name should be between 3 and 40 characters long: 'pr' (`identifier_name`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/NSRegularExpression+FilesImport.swift#L6** - Line should be 120 characters or less: currently 124 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/Runtime.swift#L38** - TODOs should be resolved (Support running from a fork?). (`todo`)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **Sources/RunnerLib/SPMDanger.swift#L53** - Don't add a space between the method name and the parentheses. (`no_space_in_method_call`)
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 652ee85e4214a7f0bce6d812b79a14bf6517ab90
</p>

**Comment by 417-72KI** [comment]
As you said (https://github.com/danger/swift/issues/382#issuecomment-706437188), SwiftLint should be executed in whole project.
> Some of them are in the JSON, and I'm in favour of ignore them, while I would like to keep the Test files still at a quality that is in line with the rest of the code if possible.

But in current, SwiftLint verifies only on `Sources`.
I'm trying to fix by unsetting `--path` property on `swiftlint`, and it should not need to set config file (.swiftlint.yml file on project root will be read).

**Comment by f-meloni** [comment]
@417-72KI what about using

```swift
let files = (danger.git.modifiedFiles + danger.git.createdFiles).filter { ($0.starts(with: "Tests") ||  $0.starts(with: "Sources")) && 
$0.fileType == .swift }

SwiftLint.lint(.files(files), inline: true)
```

?

**Comment by 417-72KI** [comment]
@f-meloni 
Here is my solution (https://github.com/danger/swift/pull/383/files#diff-01136c42fecd3d9a2f9d247c6391859f063fcf4bdb445092e14ccbc8ed02bfcfR23)

or, do you mean to make it runs twice (for `Sources` and for `Tests`) ?

**Comment by f-meloni** [comment]
@417-72KI The code I wrote is basically filtering the modified files to only include files in test and sources, but I'm totally fine with your solution. Thank you!

**Pull request merged**